src/protocol/vm/state.rs

```diff
@@ -152,11 +152,11 @@ impl VMPoolState<PreCachedDB> {
                 .contains(&Capability::ScaledPrice)
             {
                 *price_result.first().ok_or_else(|| {
-                    SimulationError::DecodingError("Spot price is not a u64".to_string())
+                    SimulationError::FatalError("Spot price is not a u64".to_string())
```

Comment:
I don't think all prices will fit into a u64... I know this is outside the scope of this PR but we should probably address it...